local-tests/setup/shiva-client.ts

```diff
@@ -162,10 +172,13 @@ export class TestnetClient {
       .then((res: Response) => {
         if (res.status === 200) {
           return res.json();
+        } else {
+          return Promise.reject(res);
```

Comment:
This is a good point, I wanted to avoid un handled promise rejections. in the case of an error on the fetch if the user was not providing their own wrappings. But I see the point. I'll make a task to refactor the shiva bindings to throw the exception rather than returning a rejected promise. 


These implementations are internal to our test suites so is not being consumed externally.